Dockerfile

```diff
@@ -10,15 +10,17 @@ RUN echo "--- Installing dependencies ---"
 WORKDIR /todo_app
 COPY pyproject.toml poetry.lock /todo_app/
 
-RUN poetry install
+#RUN poetry install
+RUN poetry config virtualenvs.create false --local && poetry install
 
 COPY ./todo_app /todo_app/todo_app
 
 FROM base as production 
 # Configure for production
 ENV FLASK_ENV=production
-ENTRYPOINT [ "poetry", "run", "gunicorn", "--bind", "0.0.0.0:5000", "todo_app.app:create_app()"]
-EXPOSE 5000
+COPY ./entrypoint.sh /todo_app/
+RUN chmod +x ./entrypoint.sh
+ENTRYPOINT ./entrypoint.sh
```

Comment:
Note that this change has just slightly broken your production container running locally -
```
production  | Skipping virtualenv creation, as specified in config file.
production  | 
production  | Error: '' is not a valid port number.
production  |
production exited with code 1
```

That is, since we now rely on the PORT environment variable being set, we should set a default or tell someone! Easiest might be to add `ENV PORT=5000` into your Dockerfile - that way a default of 5000 is set (which matches your docker-compose.yml) but Heroku can override it by passing a different value